File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    answer = 0
    q = deque()
    for idx, p in enumerate(priorities):
        q.append((p, idx))
    
    max_val = max(x[0] for x in q)
    while q:
        if (q[0][0] < max_val):
            q.append(q.popleft())
        else:
            pop_val = q.popleft()
            answer += 1
            if pop_val[1] == location: break
            if q: max_val = max(x[0] for x in q)
        
    return answer

File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 1;
        
        Deque<int[]> dq = new ArrayDeque<>();
        for(int i=0;i<priorities.length; i++) {
            dq.addLast(new int[]{i, priorities[i]}); // index, priority
        }
        
        while (!dq.isEmpty()) {
            int max = 0;
            for (int[] job : dq) max = Math.max(max, job[1]);
            
            int[] cur = dq.pollFirst();
            if (cur[1] < max) {
                dq.addLast(cur);
            } else {
                if (cur[0] == location) break;
                else answer ++;
                
            }
        }
        
        return answer;
    }
}